### PR TITLE
Additional design notes based on POC results

### PR DESCRIPTION
**Pull request by stephenfuqua**

### CODE REVIEW

File: docs/DMS/PRIMARY-DATA-STORAGE/ALTERNATIVES-DESIGNS.md [hunk 1]
```diff
@@ -1,51 +1,6 @@
-# DMS Feature: Primary Data Storage
```
Comment (stephenfuqua) [inline docs/DMS/PRIMARY-DATA-STORAGE/ALTERNATIVES-DESIGNS.md:1]: Moved design options to separate document.

File: docs/DMS/PRIMARY-DATA-STORAGE/README.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
+# DMS Feature: Primary Data Storage
+
+> [!NOTE]
+> The application architecture will have a plugin system that enables others to
+> customize the database storage. The design described below will be implemented
+> with Microsoft SQL Server (MSSQL) and PostgreSQL plugins that come out of the
+> box with the Data Management Service. Other designs could be built and
+> implemented, so long as the REST API implementation remains unchanged.
+
+## Problems to solve via DB design
```
Comment (stephenfuqua) [inline docs/DMS/PRIMARY-DATA-STORAGE/README.md:10]: There are no changes in this requirements section.

File: docs/DMS/PRIMARY-DATA-STORAGE/README.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+that we can simplify over the ODS/API by avoiding code-per-resource and instead use a generalized design
+applicable to any resource. While it may not be possible with Tanager to do this completely, we would like to
+do it whenever possible.
+
+### Performance
+
+We need to make sure that we design to meet performance goals, and test early and often to ensure that we are
+able to hit those goals.
+
+## Solution
```
Comment (stephenfuqua) [inline docs/DMS/PRIMARY-DATA-STORAGE/README.md:53]: Everything below is new.

**Review by bradbanister — CHANGES_REQUESTED** [review]
The DMS-125 branch is behind on the primary data storage documentation. I've since archived the original doc as docs/DMS/PRIMARY-DATA-STORAGE-alternatives.md. docs/DMS/PRIMARY-DATA-STORAGE.md was rewritten as Option B similar to here but with much more detailed information, and also a change to one foreign key relationship.

File: docs/DMS/PRIMARY-DATA-STORAGE/README.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+### Implementation Notes
+
+The proof-of-concept process helped uncover some details that need to guide the
+implementation:
+
+* Compared to `varbinary`, using `nvarchar` for the JSON documents results in
+  higher performance and smaller storage requirements.
+* The _partition key_ should be defined in a SQL function that takes the
+  _primary key_ value as input, so that an implementation can modify the
+  partition key without having to recompile the C# application.
```
Comment (bradbanister) [inline docs/DMS/PRIMARY-DATA-STORAGE/README.md:182]: I was thinking the number-of-partitions would be an app settings type thing, so no need to recompile. If the UUID -> partition key is fully encapsulated in a stored proc, wouldn't that mean the number-of-partition settings would have to be in some config table that gets queried a lot by the stored proc? If it's an app setting, either the C# could derive the partition key and pass it directly as a stored proc parameter, or the number-of-partitions config setting could be a parameter.

Comment (bradbanister) [reply docs/DMS/PRIMARY-DATA-STORAGE/README.md:182]: I think we may also want the partition key computation to be in C# for the query table support. We wouldn't be able to do a simple insert into a query table without knowing the partition key of the document. If it were done in a stored proc, we'd have to get the partition key returned from the DB first. Doable but more complex than just having it to begin with.

**Comment by stephenfuqua** [comment]
Closing to create a new PR with rebase.

File: docs/DMS/PRIMARY-DATA-STORAGE/README.md (hunk 4 above)
Comment (stephenfuqua) [reply docs/DMS/PRIMARY-DATA-STORAGE/README.md:182]: Thinking about this...

**Pull request closed without merging**